유저 상세 페이지 데이터 렌더링

hyokualexkwon opened the pull request:
## 작업 내용

<!-- 작업 내용을 상세하게 기술한다. -->
<!-- what 보단 why 와 how에 대해 서술한다 -->

### Server

- 한 명의 유저의 데이터를 반환하는 API를 작성했습니다.
- 유저 한명이 업로드한 비디오 리스트를 반환하는 API를 작성했습니다.

### Client

- 유저 상세 페이지를 작성했습니다.
- 비디오 리스트에 대해 스켈레톤 UI를 작성했습니다.

## 실행 방법

## 결과 첨부

![image](https://user-images.githubusercontent.com/17868599/70203757-5eebce00-1761-11ea-8aad-9c8c9738484c.png)

hyokualexkwon committed:
[server] | feat(user): 유저 상세 및 유저 비디오 리스트 api 작성

한 명의 유저 데이터와 유저의 비디오 리스트를 반환하는 api를 작성했습니다.

hyokualexkwon committed:
[server] | refactor(constants): 상수 공통화

hyokualexkwon committed:
[client] | refactor(hotlist): 핫리스트 데이터 호출 hooks 로직 수정

hyokualexkwon committed:
[client] | feat(VideoItem): 비디오 항목 스켈레톤 UI 작성

hyokualexkwon committed:
[client] | refactor(comments): 댓글 리스트 갯수 표시 로직 수정

hyokualexkwon committed:
[client] | feat(UserInfo): 유저 상세 데이터 렌더링 및 스켈레톤 작성

hyokualexkwon committed:
[client] | feat(User): 사용자 상세 데이터 호출 hook 작성

hyokualexkwon committed:
[client] | feat(User): 사용자 비디오 리스트 호출 hook 작성

hyokualexkwon committed:
[client] | feat(User): 사용자 데이터 호출 및 UI 렌더링

hyokualexkwon committed:
[server] | refactor(UserParamDto): pipe에서 사용하는 dto 공통화

hyokualexkwon committed:
[server] : refactor(User): 컨트롤러 dto 재정의

hyokualexkwon committed:
[server] | refactor(): 커밋 병합 과정에서 발생한 실수 수정

hyokualexkwon committed:
[client] | feat(loader): 공통 원형 로더 컴포넌트 작성

hyokualexkwon committed:
[client] | refactor(User): 로더 적용 및 무한 스크롤 스타일 추가

hyokualexkwon committed:
[client] | feat(Menu): 유저 관련 드롭다운 메뉴 추가

hyokualexkwon committed:
[client] | refactor(constants): 상수 변수 사용

hyokualexkwon committed:
Merge branch 'develop' into feature/user_item_20191130

Atercatus commented on packages/client/components/VideoComments/index.tsx:
```diff
@@ -40,7 +40,7 @@ const VideoComments = () => {
   return (
     <S.VideoComments>
-      <S.Title>{hasData && count !== 0 ? `${count}개의 댓글` : '댓글'}</S.Title>
+      <S.Title>{count === 0 ? '댓글' : `${count}개의 댓글`}</S.Title>
```
`!count ? ` 해도 될거같아여

hyokualexkwon replied:
👍

Atercatus commented on packages/client/components/VideoItem/styles.tsx:
```diff
@@ -110,14 +110,47 @@ export const Thumbnail = styled.div`
+
+    @media only screen and (min-width: ${BREAKPOINT}px) {
+      ${props => {
+        if (props.desktopType === 'vertical') {
```
ㅎㅎ;;

KimJye commented on packages/server/src/user/pipe/user-video-list-query.pipe.ts:
```diff
@@ -0,0 +1,35 @@
+
+  private validatePage(page: string) {
+    const parsedPage = parseInt(page, 10);
+    return !isNaN(parsedPage) && parsedPage > 0;
```
기존에 만들어둔 정규식을 사용하는것은 어떨까유~

hyokualexkwon replied:
👍

KimJye commented on packages/server/src/user/pipe/user-param.pipe.ts:
```diff
@@ -0,0 +1,23 @@
+
+  private validateId(id: string) {
+    const parsedId = parseInt(id, 10);
+    return !isNaN(parsedId);
```
여기도 정규식 어떤가요옹

KimJye commented on packages/server/src/user/pipe/user-video-list-param.pipe.ts:
```diff
@@ -0,0 +1,24 @@
+
+  private validateId(id: string) {
+    const parsedId = parseInt(id, 10);
+    return !isNaN(parsedId);
```
여기도 기존에 만들어둔 정규식을 이용하는것은어떨까용

Atercatus commented on packages/client/components/VideoItem/skeleton.tsx:
```diff
@@ -0,0 +1,50 @@
+import * as S from './styles';
+
+const VideoItemSkeleton = ({
+  mobileType = 'vertical',
```
`vertical` 상수가 있지 않나여?

Atercatus commented on packages/client/views/Hotlist/hooks.ts:
```diff
@@ -8,21 +8,26 @@ const createHotlistAction: Action = (page, period) => ({
   useEffect(() => {
     if (payload && !error) {
+      setHasData(true);
       setHasMore(payload.data.length >= 20);
```
20도 상수 변수 사용하면 좋을 거 같습니다.

Atercatus commented on packages/client/views/User/hooks.ts:
```diff
@@ -0,0 +1,76 @@
+
+export const useVideos = id => {
+  const [page, setPage] = useState(1);
+  const [sort, setSort] = useState('popular'); // popular or latest
```
😘

Atercatus commented on packages/client/views/User/hooks.ts:
```diff
@@ -0,0 +1,76 @@
+  useEffect(() => {
+    if (payload && !error) {
+      setHasData(true);
+      setHasMore(payload.data.length >= 12);
```
😘

Atercatus commented on packages/server/src/user/user.service.ts:
```diff
@@ -77,4 +81,40 @@ export class UserService {
+      .offset(offset);
+
+    if (sort === LATEST) {
+      return qb.orderBy('Video_createdAt', 'DESC').getManyAndCount();
```
map같은 걸 끼얹으면 깔끔하게 될 거 같아여

hyokualexkwon committed:
[server] | refactor(): isNaN 제거 및 이름 변경

isNaN을 사용하는 것 대신 정규표현식으로 사용하도록 변경했습니다. 또, user의 video-list임을 나타낼 수 있도록 이름을 변경했습니다.

hyokualexkwon committed:
[client] | feat(): 스켈레톤의 before content 삭제

화면에서 나타나는 경우가 있어서 제거했습니다.

riverandeye reviewed (commented):
디자이너분의 좋은 디자인 덕분에 좋은 화면이 나오는것 같아요~

존재하지 않는 사용자 id로 접속한 경우에도 계속 기다리는 것으로 보입니다.
적절하지는 않은 것 같은데, 추후에 수정하는것도 괜찮을 것 같아요.

hyokualexkwon committed:
[client] | refactor(): 조건 표현 개선

hyokualexkwon committed:
[client] | refactor(): 상수 변수 추가

hyokualexkwon committed:
Merge branch 'develop' into feature/user_item_20191130

# Conflicts:
#	packages/client/components/AppBar/index.tsx
#	packages/client/constants.ts
#	packages/client/svgs/index.tsx
#	packages/server/src/common/constants.ts
#	packages/server/src/user/user.controller.ts
#	packages/server/src/user/user.module.ts
#	packages/server/src/user/user.service.ts
#	packages/server/src/video/pipe/comments-query-pipe.ts
#	packages/server/src/video/video.service.ts

hyokualexkwon committed:
[server] | refactor(): 상대 경로로 변경
